src/components/project/overview/description.vue

```diff
@@ -23,53 +23,155 @@ except according to the terms contained in the LICENSE file.
+  // edit link if they are a manager and can update the project.
+  // If the description is blank and they don't have permission,
+  // nothing will be shown in this description component.
+  canUpdate: {
+    type: Boolean
+  }
+});
+
+const { projectPath } = useRoutes();
+
+const emptyDescription = computed(() => !props.description);
+
+const markdownRef = ref(null);
+
+const targetHeight = ref(null);
+
+const getNumberOfLines = (box, offsetHeight, lineHeight) => {
+  const contentHeight = offsetHeight - box.paddingTop - box.paddingBottom;
+
+  return Math.round(contentHeight / lineHeight);
+};
+
+/**
+ * Returns the desired height for the element based on the number of text lines in the element and
+ * number of lines to show on the UI. If number of lines in the element is less than or equal to
+ * allowedLines then `null` is returned.
+ */
+const getTargetHeight = (element, allowedLines) => {
+  let linesCount = 0;
+  let runningHeight = 0;
+
+  for (let i = 0; i < element.children.length; i += 1) {
+    const child = element.children[i];
+
+    // Get/compute required inputs to determine the target height
+    const style = window.getComputedStyle(child);
+    const box = styleBox(style);
+    let { lineHeight } = box;
+    if (child.tagName === 'IMG') {
+      lineHeight = 20;
     }
+    const numberOfLines = getNumberOfLines(box, child.offsetHeight, lineHeight);
+
+    // Adjustment for margin collapsing
+    const previousMarginBottom = element.children[i - 1]
+      ? styleBox(window.getComputedStyle(element.children[i - 1])).marginBottom
+      : 0;
+    const adjustedMarginTop = Math.max(box.marginTop - previousMarginBottom, 0);
+    const totalHeight = adjustedMarginTop + child.offsetHeight + box.marginBottom;
+
+    // we have exceeded the allowed number of lines
+    // OR We have reached exact allowed number lines and there are more
+    // elements after this one
+    if (linesCount + numberOfLines > allowedLines ||
+          (linesCount + numberOfLines === allowedLines && (i + 1) < element.children.length)) {
+      const linesToShowOfThisNode = allowedLines - linesCount - 0.5;
+      return runningHeight +
+        adjustedMarginTop +
+        box.paddingTop +
+        (linesToShowOfThisNode * lineHeight);
+    }
+
+    // Update running totals
+    linesCount += numberOfLines;
+    runningHeight += totalHeight;
+  }
+  return null;
+};
+
+const setMarkdownHeight = () => {
+  if (!markdownRef.value) return;
+
+  const expanded = targetHeight.value && markdownRef.value.$el.style.height === 'auto';
+
+  targetHeight.value = getTargetHeight(markdownRef.value.$el, 4);
+
+  if (!expanded && targetHeight.value) {
+    markdownRef.value.$el.style.height = `${targetHeight.value}px`;
+  }
+};
+
+onMounted(() => {
+  // nextTick is good enough for the calculation being done here. I don't see
+  // any screen flicker. Maybe related: https://github.com/vuejs/vue/issues/9200.
+  // Additionally Without nextTick offsetHeight is always 0.
```

Comment:
I bet this is because there's a `v-show` on `<page-head>`. Normally I wouldn't expect `nextTick()` to be needed in `onMounted()`, since the element is in the DOM at that point. In any case, I think it makes sense the way it is. :+1: